resolwe/flow/models.py

```diff
@@ -583,6 +594,81 @@ def validation_schema(name):
     return json.loads(schema.replace('{{FIELD}}', field_schema).replace('{{PARENT}}', '/field'))
 
 
+TYPE_SCHEMA = validation_schema('type')
+
+
+def validate_schema(instance, schema):
+    """Check if DictField values are consistent with our data types.
+
+    Perform basic JSON schema validation and our custom validations:
+
+      * check that required fields are given
+      * check if ``basic:file:`` and ``list:basic:file`` fields matches
+        regex given in schema (only if ``validate_regex`` is defined in
+        schema for coresponding fields)
+      * check that referenced ``Data`` objects (in ``data:<data_type>``
+        and  ``list:data:<data_type>`` fields) exists and are of type
+        ``<data_type>``
+      * check that referenced ``Storage`` objects (in ``basic:json``
+        fields) exists
+
+    :param list instance: Instance to be validated
```

Comment:
Yes, it works for simple types. Here are some examples: http://www.sphinx-doc.org/en/stable/domains.html#info-field-lists